vermouth/processors/do_links.py

```diff
@@ -216,10 +220,24 @@ def match_link(molecule, link):
     GM = nx.isomorphism.GraphMatcher(molecule, link, node_match=_atoms_match)
 
     raw_matches = GM.subgraph_isomorphisms_iter()
+    avoid_doubles = []
     for raw_match in raw_matches:
         # raw_match: mol -> link
         # rev_raw_match: link -> mol
         rev_raw_match = {value: key for key, value in raw_match.items()}
+        
+        # When the 'other' operator (*) is used, links involving
+        # same atoms from different residues will be doubled over
+        # permutations. This needs to be avoided.
+        unique = {a.lstrip('*') if isinstance(a, str) else a for a in raw_match.values()}
+        if len(unique) == 1:
```

Comment:
This will only work for, e.g. a Link `SG *SG`, and not for `CB SG *SG *CB`. If you want to implement the latter option, you should also take connectivity into account I think though, so that'll become messy.
Actually, what we need/want is to be able to decide whether or not a link is symmetric (automorphism!), but then we're back at the discussion at #57.

Reply:
This would also be solved with a resid class, making * mean 'different molecule'